freetype/build.zig

```diff
@@ -0,0 +1,138 @@
+const std = @import("std");
+
+const ft_root = thisDir() ++ "/upstream/freetype";
+
+fn thisDir() []const u8 {
+    return std.fs.path.dirname(@src().file) orelse ".";
+}
+
+pub fn buildFreeType(b: *std.build.Builder, mode: std.builtin.Mode, target: std.zig.CrossTarget, root_path: []const u8, custom_config_path: ?[]const u8) !*std.build.LibExeObjStep {
```

Comment:
@alichraghi I'm hoping we can change this signature to match more the `glfw.link`, `gpu_dawn.link`, etc. pattern I am using in other libraries - giving an `Options` struct. Like this: https://github.com/hexops/mach/blob/f64595393b202e54f097c22e839d33db67528c6a/glfw/build.zig#L49

Thoughts?

Reply:
what options we are gonna have?